File: GTP2.5/gtp2.cpp
#include<iostream>
#include<string.h>

using namespace std;

void parse(string names[], string data[], int ages[]) {
        int indx = 0;

        for(short i = 0; i < 6; i++) {
                if(ages[i] == 18 or ages[i] == 20 or ages[i] == 22) {
                        data[indx] = names[i];
                        indx++;
                }
        }
}

void input(string names[], int ages[]) {
        for(short i = 0; i < 6; i++) {
                cout << "Ingrese el nombre de la persona " << i+1 << ": ";
                cin >> names[i];

                cout << "Ingrese su edad: ";
                cin >> ages[i];
        }
}

void print(string data[]) {
        cout << "Las personas que tienen 18, 20 o 22 años son:\n";

        for(short i = 0; i < 6; i++) {
                cout << data[i] << "\n";
        }
}

int main() {
        string names[6] = {"Juan","Laura","Emilia","Julian","Paola","Lorenzo"}, data[6];
        int ages[6] = {20,15,26,22,18,23};
        //string names[6], data[6];
        //int ages[6];

        //input(names, ages);
        
        parse(names, data, ages);

        print(data);

        return 0;
}

File: GTP2.5/gtp2.py
def almacenar(nombres:list[str], edades, nom):
        for i in range(6):
                if edades[i] == 18 or edades[i] == 20 or edades[i] == 22:
                        nom.append(nombres[i])

def main():
        nombres:list[str] = []
        nom:list[str] = []
        edades:list[int] = []

        for i in range(6):
                nombres.append(input(f"Ingrese el nombre de la persona {i+1}: "))
                edades.append(int(input("Ingrese su edad: ")))
                print()              

        almacenar(nombres, edades, nom)

        print(f"\nLas personas que tienen 18, 20 o 22 años son: {nom}")

if __name__ == '__main__':
        main()
